### PR TITLE
Allow historical storage data to be recorded - For use during migrations

### LINKED ISSUE #2451: Allow historical storage data to be recorded for environments
Currently the `addOrUpdateEnvironmentStorage` mutation only accepts the storage data to record, and it hardcodes the date to be the time of submission. When moving projects between clusters, we need a way to move historical storage data also.

Two possible solutions:
1. Update the existing mutation to accept a date (it can be optional for backwards compatibility)
2. Add a new mutation specifically for this use case

The advantage to (1) is simplicity. The advantage to (2) is the API client wouldn't need to load all data, then loop through it all and call a bunch of API calls.

### PR DESCRIPTION
**Pull request by justinlevi**
# Checklist

- [x] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [x] PR title is ready for changelog and subsystem label(s) applied

Explain the **details** for making this change. What existing problem does the pull request solve?

This PR adds the ability to allow for historical storage data to be recorded. This is useful during migrations. Adds an additional `updated` field on the mutation. 

```
mutation {
  addOrUpdateEnvironmentStorage(input:{
    environment:20013, 
    persistentStorageClaim:"BananaSmoothiesRule", 
    bytesUsed:1234,
    updated:"2020-12-10"
  }){
    id
    environment{
      name
    }
    persistentStorageClaim
    bytesUsed
    updated
  }
  ```

# Closing issues

Put `closes #2451`

### CODE REVIEW

File: services/api/src/resources/environment/resolvers.ts [hunk 1]
```diff
@@ -344,21 +344,30 @@ export const addOrUpdateEnvironment: ResolverFn = async (
...
       CALL CreateOrUpdateEnvironmentStorage(
         :environment,
         :persistent_storage_claim,
-        :bytes_used
+        :bytes_used,
+        :updated
       );
     `,
   );
+  console.log("🚀 ~ file: resolvers.ts ~ line 368 ~ prep", prep(input))
```
Comment (Schnitzel) [inline services/api/src/resources/environment/resolvers.ts]: let' remove that :)

File: services/api/src/resources/environment/resolvers.ts [hunk 2]
```diff
@@ -344,21 +344,30 @@ export const addOrUpdateEnvironment: ResolverFn = async (
...
 export const addOrUpdateEnvironmentStorage: ResolverFn = async (
   root,
-  { input },
+  { input: unformattedInput },
   { sqlClient, hasPermission },
 ) => {
   await hasPermission('environment', 'storage');
 
+  const d = new Date();
+  const now = d.toISOString().split('T')[0]+' ' + d.toTimeString().split(' ')[0];
```
Comment (Schnitzel) [inline services/api/src/resources/environment/resolvers.ts]: this seems very hacky, is that the correct way to generate a specific formatted string?

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: services/api/src/resources/environment/resolvers.ts (hunk 1 above)
Comment (justinlevi) [reply services/api/src/resources/environment/resolvers.ts]: fixed

File: services/api/src/resources/environment/resolvers.ts (hunk 2 above)
Comment (justinlevi) [reply services/api/src/resources/environment/resolvers.ts]: Yeah, you're probably right. I've updated and reused a formatter function we use elsewhere.

**Review by Schnitzel — APPROVED** [review]

**Review by rocketeerbkw — APPROVED** [review]

**Pull request merged**